packages/components/src/color-picker/saturation.js

```diff
@@ -0,0 +1,186 @@
+import { TAB } from '@wordpress/keycodes';
+import { withInstanceId } from '@wordpress/compose';
+
+/**
+ * Internal dependencies
+ */
+import { calculateSaturationChange } from './utils';
+import KeyboardShortcuts from '../keyboard-shortcuts';
+
+export class Saturation extends Component {
+	constructor( props ) {
+		super( props );
+
+		this.throttle = throttle( ( fn, data, e ) => {
+			fn( data, e );
+		}, 50 );
+
+		this.container = createRef();
+		this.saturate = this.saturate.bind( this );
+		this.brighten = this.brighten.bind( this );
+		this.handleChange = this.handleChange.bind( this );
+		this.handleMouseDown = this.handleMouseDown.bind( this );
+		this.handleMouseUp = this.handleMouseUp.bind( this );
+	}
+
+	componentWillUnmount() {
+		this.throttle.cancel();
+		this.unbindEventListeners();
+	}
+
+	saturate( amount = 0.01 ) {
+		const { hsv, onChange = noop } = this.props;
+		const intSaturation = clamp(
+			hsv.s + Math.round( amount * 100 ),
+			0,
+			100
+		);
+		const change = {
+			h: hsv.h,
+			s: intSaturation,
+			v: hsv.v,
+			a: hsv.a,
+			source: 'rgb',
+		};
+
+		onChange( change );
+	}
+
+	brighten( amount = 0.01 ) {
+		const { hsv, onChange = noop } = this.props;
+		const intValue = clamp(
+			hsv.v + Math.round( amount * 100 ),
+			0,
+			100
+		);
+		const change = {
+			h: hsv.h,
+			s: hsv.s,
+			v: intValue,
+			a: hsv.a,
+			source: 'rgb',
+		};
+
+		onChange( change );
+	}
+
+	handleChange( e ) {
+		const { onChange = noop } = this.props;
+		const change = calculateSaturationChange( e, this.props, this.container.current );
+		this.throttle( onChange, change, e );
+	}
+
+	handleMouseDown( e ) {
+		this.handleChange( e );
+		window.addEventListener( 'mousemove', this.handleChange );
+		window.addEventListener( 'mouseup', this.handleMouseUp );
+	}
+
+	handleMouseUp() {
+		this.unbindEventListeners();
+	}
+
+	preventKeyEvents( event ) {
+		if ( event.keyCode === TAB ) {
+			return;
+		}
+		event.preventDefault();
+	}
+
+	unbindEventListeners() {
+		window.removeEventListener( 'mousemove', this.handleChange );
+		window.removeEventListener( 'mouseup', this.handleMouseUp );
+	}
+
+	render() {
+		const { hsv, hsl, instanceId } = this.props;
+		const pointerLocation = {
+			top: `${ -( hsv.v ) + 100 }%`,
+			left: `${ hsv.s }%`,
+		};
+		const shortcuts = {
+			up: () => this.brighten(),
+			'shift+up': () => this.brighten( 0.1 ),
+			pageup: () => this.brighten( 1 ),
+			down: () => this.brighten( -0.01 ),
+			'shift+down': () => this.brighten( -0.1 ),
+			pagedown: () => this.brighten( -1 ),
+			right: () => this.saturate(),
+			'shift+right': () => this.saturate( 0.1 ),
+			end: () => this.saturate( 1 ),
+			left: () => this.saturate( -0.01 ),
+			'shift+left': () => this.saturate( -0.1 ),
+			home: () => this.saturate( -1 ),
+		};
+
+		/* eslint-disable jsx-a11y/no-static-element-interactions */
+		return (
+			<KeyboardShortcuts shortcuts={ shortcuts }>
+				<div
```

Comment:
The `role="application"` I've suggested should be set here, together with a meaningful `aria-label`.